### PR TITLE
[feature][txn] Fix individual ack batch message with transaction abort redevlier duplicate messages

### PR DESCRIPTION
**Pull request by congbobo184**
### Motivation
If individual ack batch message with transaction and abort this transaction, we will redeliver this message. but this batch message some bit sit are acked by another transaction and re consume this bit sit will produce `TransactionConflictException`, we don't need to redeliver this bit sit witch is acked by another transaction.

if batch have batch size 5

1. txn1 ack 0, 1     the ackSet is   00111
2. txn2 ack 2 3 4 the ack Set is  11000
3. abort txn2 redeliver this position is 00111
4. but now we don't filter txn1 ackSet so redeliver this position bitSet is 111111
### Modifications
When filter the message we should filter the bit sit witch is real ack or in pendingAck state
### Verifying this change
add the test

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

- [x] `doc-not-needed`

### CODE REVIEW

**Review by liangyepianzhou — APPROVED** [review]

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/PositionAckSetUtil.java [hunk 1]
```diff
@@ -46,12 +46,18 @@ public static void andAckSet(PositionImpl currentPosition, PositionImpl otherPos
         if (currentPosition == null || otherPosition == null) {
             return;
         }
-        BitSetRecyclable thisAckSet = BitSetRecyclable.valueOf(currentPosition.getAckSet());
-        BitSetRecyclable otherAckSet = BitSetRecyclable.valueOf(otherPosition.getAckSet());
+        currentPosition.setAckSet(andAckSet(currentPosition.getAckSet(), otherPosition.getAckSet()));
+    }
+
+    //This method is do `or` operation for ack set
```
Comment (Jason918) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/PositionAckSetUtil.java:52]:
```suggestion
    //This method is do `and` operation for ack set
```
Typo?

**Review by Jason918 — APPROVED** [review]
LGTM

**Comment by congbobo184** [comment]
This pr is not merged for now, so merging this modification may result in lost messages

**Comment by mattisonchao** [comment]
> This pr is not merged for now, so merging this modification may result in lost messages
Hi, @congbobo184 
I don't understand why we merge this PR will cause messages to be lost?

**Comment by Jason918** [comment]
@congbobo184 Should we close this PR?

**Comment by congbobo184** [comment]
@Jason918 don't close this PR, I will deal with it

**Comment by thetumbled** [comment]
Isn't there a problem of message loss  with this PR?

**Comment by codelipenghui** [comment]
We should carefully to cherry-pick this fix to release branch.

**Comment by mattisonchao** [comment]
@thetumbled I will do testing this week to check the problem you mentioned.

**Comment by coderzc** [comment]
@congbobo184 Can you help cherry-pick it to branch 2.9?

**Comment by michaeljmarshall** [comment]
As discussed on the mailing list https://lists.apache.org/thread/w4jzk27qhtosgsz7l9bmhf1t7o9mxjhp, there is no plan to release 2.9.6, so I am going to remove the release/2.9.6 label

**Pull request merged**